### ISSUE TITLE
error on dynamic table refresh during

### ISSUE THREAD

**Issue by freylafert** [Snowflake-Labs/terraform-provider-snowflake#2484]
### Terraform CLI and Provider Versions

Terraform v1.5.5
on darwin_arm64

Provider version 0.85.0

### Terraform Configuration

```terraform
resource "snowflake_dynamic_table" "dynamic_table" {
  provider   = snowflake.account_admin
  for_each   = var.dynamic_tables
  database   = var.snowflake_database_name
  warehouse  = var.snowflake_warehouse_name
  schema     = snowflake_schema.collection.name
  name       = each.value.name
  comment    = each.value.name
  or_replace = true
  query      = templatefile("${path.module}/${each.value.sql}", { DB = var.snowflake_database_name, SCHEMA = snowflake_schema.collection.name, IP_INFO_PRIVATE_SHARING = var.ip_info_private_sharing })
  target_lag {
    maximum_duration = var.env == "ci" || var.env == "dev" ? "1 minute" : each.value.target_lag
  }
}
```

### Expected Behavior

expected terraform to be applied

### Actual Behavior

Error: too many matches found. There is no way of getting ONLY the 'query' used to create the dynamic table from Snowflake. We try to get it from the whole creation statement but there may be cases where it fails. Please submit the issue on Github (refer to #2329)

#2329

### Steps to Reproduce

1. `terraform apply`

### How much impact is this issue causing?

High

**Comment by sfc-gh-asawicki** [comment]
Hey @freylafert. Thanks for reaching out to us.

Can you share the query you want to run? I suspect that it has multiple ` SELECT AS ` in it. This is the limitation in the current provider version which is caused by how Snowflake returns the query. You can check more details in #2329.

There is also an open PR #2438 that will try to solve such problems. We should be able to release it as a part of v0.86.0 (this Wednesday/Thursday).

**Comment by freylafert** [comment]
Yes, we have statements like this in the query:
`WITH unique_accounts_per_day_on_ip AS`

**Comment by sfc-gh-asawicki** [comment]
Great, then it will be a good test for the change from #2438 (please check it, after we release v0.86.0.

**Comment by sfc-gh-asawicki** [comment]
Hey @freylafert. We have released 0.86.0 yesterday. Could you check if this solves your issue, please?

**Comment by freylafert** [comment]
Yes, the issue was resolved, closing the ticket.
